### ISSUE TITLE
Some strangeness in the PWM interface

### ISSUE THREAD

**Issue by sakian** [atsamd-rs/atsamd#584]
I've been using the PWM interface for a bit now and there's some strangeness in it. I just want to make sure I'm not missing anything before I potentially look to create a PR. Currently, I'm using the ATSAMD51J18A chip and trying to drive two PWM outputs on PA04 and PA05 using TC0. I think this should be possible, but running into a number of issues that made me look closer at the HAL implementation. Weird things I think I'm seeing are:

- **Difference between `thumbv6m` and `thumbv7em`**: `thumbv7em` allows you to give it a pin, where `thumbv6m` doesn't. Seems like `thumbv6m` should be updated to take a pinout as well (talked about in #245).
- **`thumbv7em` TCx implements `PwmPin` and not `Pwm`**: The TCx can control two pins, so I think these should implement `Pwm`?
- **Can only give it a single pin**: Multiple pins can be driven from a single PWM (is this true for all chips?) so it makes sense that we should be able to give it multiple pins. Right now, I create pinouts for each pin to initialize them, but can only pass one to the PWM.
- **Not all supported PWM pins are defined in the HAL**: So, for what I'm working on now, there is a `TC0Pinout` defined for `PA05`, but none for `PA04` even though it's supported by the chip.

**Comment by bradleyharden** [comment]
@sakian, yes, there are a few APIs in the HAL that could probably use an overhaul. I've notice the inconsistencies in the PWM modules myself.

If you want to tackle this, I would suggest making a mini-RFC first, outlining what you plan to do. I think that would reduce any wasted effort and also make it easier for reviewers to understand the approach without dealing with all the complications of the actual implementation.
